### PR TITLE
Angular - Datatable Loading Indicator Updates

### PR DESCRIPTION
**Pull request by sumeyyeKurtulus**
### Description

Updates the `ngx-datatable` implementation bu modifying the loading indicator (progress) to show a customized spinner.

LoadingIndicator

### Checklist

- [X] I fully tested it as developer / designer and created unit / integration tests

### How to test it?

- You can run `copy-to:app` command and test the tables as demonstrated in the record.

### CODE REVIEW

**Review by masum-ulu — CHANGES_REQUESTED** [review]
I'll review after request-changes are applied

File: npm/ng-packs/packages/components/extensible/src/lib/components/extensible-table/extensible-table.component.ts [hunk 1]
```diff
@@ -103,13 +112,11 @@ export class ExtensibleTableComponent<R = any> implements OnChanges {
...
-  private config = inject(ConfigStateService);
-  entityPropTypeClasses = inject(ENTITY_PROP_TYPE_CLASSES);
-  #injector = inject(Injector);
-  getInjected = this.#injector.get.bind(this.#injector);
-  permissionService = this.#injector.get(PermissionService);
+  subscriptionService = this.#injector.get(SubscriptionService);
+  listService = this.#injector.get(ListService);
+  routerWaitService = this.#injector.get(RouterWaitService);
+  cdRef = this.#injector.get(ChangeDetectorRef);
+  isLoading = true;
```
Comment (masum-ulu) [inline npm/ng-packs/packages/components/extensible/src/lib/components/extensible-table/extensible-table.component.ts]: These all variables still need to provide here ?

File: npm/ng-packs/packages/theme-shared/src/lib/components/spinner/spinner.component.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { Component } from '@angular/core';
+
+@Component({
+  exportAs: 'spinner',
+  selector: 'abp-spinner',
+  standalone: true,
+  imports: [],
```
Comment (masum-ulu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/spinner/spinner.component.ts]: Don't need to define this

File: npm/ng-packs/packages/theme-shared/src/lib/components/spinner/spinner.component.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Component } from '@angular/core';
+
+@Component({
+  exportAs: 'spinner',
```
Comment (masum-ulu) [inline npm/ng-packs/packages/theme-shared/src/lib/components/spinner/spinner.component.ts]: Why we need exportAs like that ?

File: npm/ng-packs/packages/theme-shared/src/lib/directives/ngx-datatable-list.directive.ts [hunk 4]
```diff
@@ -7,30 +7,40 @@ import {
...
   protected readonly table = inject(DatatableComponent);
   protected readonly cdRef = inject(ChangeDetectorRef);
   protected readonly destroyRef = inject(DestroyRef);
   protected readonly localizationService = inject(LocalizationService);
   protected readonly ngxDatatableMessages = inject(NGX_DATATABLE_MESSAGES, { optional: true });
+  protected readonly routerWaitService = inject(RouterWaitService);
+  protected readonly viewContainerRef = inject(ViewContainerRef);
+  protected readonly renderer = inject(Renderer2);
+
+  protected _loading = true;
```
Comment (masum-ulu) [inline npm/ng-packs/packages/theme-shared/src/lib/directives/ngx-datatable-list.directive.ts]: We've list service already, use inside loading prop of it instead using local var. Use inside of the list service for all _loading variable

File: npm/ng-packs/packages/components/extensible/src/lib/components/extensible-table/extensible-table.component.ts (hunk 1 above)
Comment (sumeyyeKurtulus) [reply npm/ng-packs/packages/components/extensible/src/lib/components/extensible-table/extensible-table.component.ts]: Seems like I forgot to remove and did so 🫡

File: npm/ng-packs/packages/theme-shared/src/lib/components/spinner/spinner.component.ts (hunk 2 above)
Comment (sumeyyeKurtulus) [reply npm/ng-packs/packages/theme-shared/src/lib/components/spinner/spinner.component.ts]: You are right, I have removed them

File: npm/ng-packs/packages/theme-shared/src/lib/directives/ngx-datatable-list.directive.ts (hunk 4 above)
Comment (sumeyyeKurtulus) [reply npm/ng-packs/packages/theme-shared/src/lib/directives/ngx-datatable-list.directive.ts]: I have also removed the extra check that I add for `RouterWaitService`

**Review by masum-ulu — CHANGES_REQUESTED** [review]
This PR will be approved after docs added

**Review by masum-ulu — APPROVED** [review]

**Comment by sumeyyeKurtulus** [comment]
Hello @masum-ulu I have handled the catchError inside the `streamCreatorCallback` function and updated the tests accordingly. I did not see anything broken after all, fyi 

https://github.com/user-attachments/assets/12112f04-2b03-4c4f-add4-1f8d5803502f

**Comment by masum-ulu** [comment]
### There are some changes that we need to do
- Set localized empty message 
- Set requestStatus to idle when get any error or after success step
- When we get error display on empty message or push a notification.

**Review by masum-ulu — CHANGES_REQUESTED** [review]
We might set idle requestStatus after request. If not need it I'll aprove @sumeyyeKurtulus

File: npm/ng-packs/packages/theme-shared/src/lib/directives/ngx-datatable-list.directive.ts [hunk 5]
```diff
@@ -55,32 +62,88 @@ export class NgxDatatableListDirective implements OnChanges, OnInit, DoCheck {
     this.table.offset = page;
   }
 
+  protected subscribeToLoadingState() {
```
Comment (masum-ulu) [inline npm/ng-packs/packages/theme-shared/src/lib/directives/ngx-datatable-list.directive.ts]: We can change method name to `subscribeToRequestStatus` might be better

**Review by masum-ulu — CHANGES_REQUESTED** [review]

**Review by masum-ulu — APPROVED** [review]

**Review by oykuermann — CHANGES_REQUESTED** [review]
I can not test the PR:
App template:
image
App-pro:
image

**Review by oykuermann — APPROVED** [review]

**Pull request merged**